Wikipedia/Code/PageEditorViewController.swift

```diff
@@ -0,0 +1,236 @@
+        
+        NSLayoutConstraint.activate([leadingConstraint, trailingConstraint, topConstraint])
+    }
+    
+    private func loadWikitext() {
+        fetcher.fetchSection(with: sectionID, articleURL: pageURL) {  [weak self] (result) in
+            DispatchQueue.main.async { [weak self] in
+                
+                guard let self else {
+                    return
+                }
+                
+                switch result {
+                case .failure(let error):
+                    print(error)
+                case .success(let response):
+                    self.addChildEditor(wikitext: response.wikitext)
+                }
+            }
+        }
+    }
+    
+    private func addChildEditor(wikitext: String) {
+        
+        let viewModel = WKSourceEditorViewModel(configuration: .full, initialText: wikitext)
+        let sourceEditor = WKSourceEditorViewController(viewModel: viewModel, delegate: self)
+        
+        addChild(sourceEditor)
+        sourceEditor.view.translatesAutoresizingMaskIntoConstraints = false
+        
+        view.addSubview(sourceEditor.view)
+        
+        let top = view.safeAreaLayoutGuide.topAnchor.constraint(equalTo: sourceEditor.view.topAnchor)
+        let bottom = view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: sourceEditor.view.bottomAnchor)
+        let leading = view.safeAreaLayoutGuide.leadingAnchor.constraint(equalTo: sourceEditor.view.leadingAnchor)
+        let trailing = view.safeAreaLayoutGuide.trailingAnchor.constraint(equalTo: sourceEditor.view.trailingAnchor)
+        
+        NSLayoutConstraint.activate([
+            top,
+            bottom,
+            leading,
+            trailing
+        ])
+        
+        sourceEditor.didMove(toParent: self)
+        self.sourceEditor = sourceEditor
+        self.editorTopConstraint = top
+    }
+    
+    private func showFocusNavigationView() {
+        navigationController?.setNavigationBarHidden(true, animated: false)
+        editorTopConstraint.constant = -focusNavigationView.frame.height
+        focusNavigationView.isHidden = false
+        
+    }
+    
+    private func hideFocusNavigationView() {
+        editorTopConstraint.constant = 0
+        focusNavigationView.isHidden = true
+        navigationController?.setNavigationBarHidden(false, animated: false)
+    }
+}
+
+// MARK: - Themeable
+
+extension PageEditorViewController: Themeable {
+    func apply(theme: Theme) {
+        guard isViewLoaded else {
+            return
+        }
+        
+        navigationItemController.apply(theme: theme)
+        view.backgroundColor = theme.colors.paperBackground
+    }
+}
+
+// MARK: - WKSourceEditorViewControllerDelegate
+
+extension PageEditorViewController: WKSourceEditorViewControllerDelegate {
+    func sourceEditorViewControllerDidTapFind(sourceEditorViewController: Components.WKSourceEditorViewController) {
```

Comment:
The `Components.` namespace prefix is unneeded here.

Reply:
Done!